src/sentry/integrations/tasks/sync_status_inbound.py

```diff
@@ -213,6 +225,13 @@ def sync_status_inbound(
         "integration_id": integration_id,
     }
     if action == ResolveSyncAction.RESOLVE:
+        # Check if the group was recently resolved and we should skip the request
+        # Avoid resolving the group in-app and then re-resolving via the integration webhook
+        # which would override the in-app resolution
+        for group in affected_groups:
+            if group.status == GroupStatus.RESOLVED and group_was_recently_resolved(group):
+                affected_groups.remove(group)
```

Comment:
Is it safe to remove items from a list you're iterating through? I always thought of this as bad practice but I'm not sure how Python handles this.

Reply:
Just tested this, and removing items in place can result in skipping over list items unintentionally. Verified this behavior with the example from this SO post: https://stackoverflow.com/questions/4960968/how-to-safely-remove-elements-from-a-list-in-python

Reply:
whoops i remember seeing that before. python really has that nice .remove function that is so tempting to use